File: TableRW.Epplus/Read/ExcelReader.cs

using OfficeOpenXml;
using TableRW.Read.I;
using TableRW.Read.I.Epplus;

namespace TableRW.Read.Epplus;

public class ExcelReader<TEntity> : ExcelReaderImpl<IContext<ExcelWorksheet, TEntity>> { }

public class ExcelReader<TEntity, TData> : ExcelReaderImpl<IContext<ExcelWorksheet, TEntity, TData>> { }

File: TableRW.Epplus.Fs/Read/ExcelReader.fs
namespace TableRW.Read.Epplus.Fs

open TableRW.Read.I
open TableRW.Read.I.Epplus.Fs
open OfficeOpenXml
open System.Collections.Generic

type ExcelReader<'TEntity>() =
    inherit ExcelReaderImpl<IContext<ExcelWorksheet, 'TEntity>>()

type ExcelReader<'TEntity, 'TData>() =
    inherit ExcelReaderImpl<IContext<ExcelWorksheet, 'TEntity, 'TData>>()

//[<AutoOpen>]
//module SystemEx2 =
//    let FilterMap seq =
//        seq |> Seq.filter fst |> Seq.map snd
    
//    type IDictionary<'k, 'v> with 
//        member dic.GetValueOr (k, orValue) =
//            match dic.TryGetValue(k) with
//            | (true, value) -> value
//            | _ -> orValue